pennylane/transforms/split_tape.py

```diff
@@ -200,40 +202,42 @@ def split_tape(tape: QuantumTape, group=True):
                 else:
                     idxs_coeffs_dict[o_m.hash].append((idx, coeff))
         else:
-            coeff = None
             if isinstance(obs, SProd) and isinstance(m, ExpectationMP):
                 coeff = obs.scalar
                 m = qml.expval(obs.base)
 
             if m.hash not in measurements_dict:
                 measurements_dict[m.hash] = m
-                idxs_coeffs_dict[m.hash] = [(idx, coeff)]
+                idxs_coeffs_dict[m.hash] = [(idx, None)]
```

Comment:
Why this change is necessary?

Reply:
It's not necessary! But a small code improvement (`coeff` was set to `None` before the `SProd` check).